### PR TITLE
CommonBranch - Added PageObjects support;

### PR DESCRIPTION
**Pull request by VladimirPetrat**

### CODE REVIEW

File: src/main/java/com/softserveinc/ita/rozetka/page_objects/BasketPage.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+    public void pushPlusItem() {
+        $x("(//*[@class='button button_color_white button_size_medium cart-counter__button'])[2]").click();
+    }
+
+    public void pushMinusItem() {
+        $x("(//*[@class='button button_color_white button_size_medium cart-counter__button'])[1]").click();
+    }
+
+    public int getOrderItemPrice() {
```
Comment (toleksyn) [inline src/main/java/com/softserveinc/ita/rozetka/page_objects/BasketPage.java]: getOrderItemPrice(1)

File: src/main/java/com/softserveinc/ita/rozetka/page_objects/FooterPage.java [hunk 2]
```diff
@@ -0,0 +1,4 @@
+package com.softserveinc.ita.rozetka.page_objects;
+
+public class FooterPage {
+}
```
Comment (toleksyn) [inline src/main/java/com/softserveinc/ita/rozetka/page_objects/FooterPage.java]: ??

File: src/main/java/com/softserveinc/ita/rozetka/page_objects/HomePage.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        return new CategoryPage();
+    }
+
+    public CategoryPage openCategoryPageFromLeftSidebar(String pageCategory) {
+        $x(String.format("//a[@class = 'menu-categories__link' and contains(text(), '%s')]", pageCategory))
+                .click();
+        $x("//button[@class='search-form__microphone']")
+                .click();
+        return new CategoryPage();
+    }
```
Comment (toleksyn) [inline src/main/java/com/softserveinc/ita/rozetka/page_objects/HomePage.java:16]: getSidebar().openCategory()

File: src/main/java/com/softserveinc/ita/rozetka/page_objects/ProductPage.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+    public ElementsCollection getListOfProductTabs() {
+        return Selenide.$$x("//a[@class='tabs__link']");
+    }
+
+    public SelenideElement getViewedProduct(int number) {
+        return $x(String.format("(//section[@class='recently-viewed']//a[@class='lite-tile__title'])[%d]", number));
+    }
+
+    public CategoryPage returnToCategoryPage(int index) {
```
Comment (toleksyn) [inline src/main/java/com/softserveinc/ita/rozetka/page_objects/ProductPage.java]: name, not index

File: src/main/java/com/softserveinc/ita/rozetka/page_objects/ProductPage.java [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    }
+
+    public BasketPage putProductOpenBasket() {
+        $x("(//span[@class='buy-button__label'])[1]").hover().click();
+        return new BasketPage();
+    }
+      
+      public ProductPage openProductTabByName(String tabName) {
+        $x(String.format("//a[@class='tabs__link' and contains(text(),'%s')]", tabName)).click();
+        return new ProductPage();
```
Comment (toleksyn) [inline src/main/java/com/softserveinc/ita/rozetka/page_objects/ProductPage.java]: return this

File: src/main/java/com/softserveinc/ita/rozetka/page_objects/ProductPage.java [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+    public String getCharacteristicDescriptionByIndex(int index) {
+        return $x(String.format("(//ul[@class='characteristics-full__sub-list'])[%d]", index)).text();
+    }
+
+    public String getProductTabsTitle(){
+        return $x("//h2[@class='product-tabs__heading']").text();
+    }
+
+    public List<SelenideElement> getCommentsList(){
+        return $$x("//div[@class='comment']");
```
Comment (toleksyn) [inline src/main/java/com/softserveinc/ita/rozetka/page_objects/ProductPage.java]: this won't be enough, no wait here

File: src/test/java/com/softserveinc/ita/rozetka_test/RozetkaProductNavigationTest.java [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+    public void openHomepage() {
+        rozetkaHomePage = new HomePage().openHomePage();
+    }
+
+    @Test
+    public void testSelectProductByCatalog() {
+        CategoryPage catalogCategoryItem = rozetkaHomePage
+                .openCategoryPageFromLeftSidebar("Ноутбуки");
+        Assert.assertTrue(catalogCategoryItem.getPageTitle().contains("Комп'ютери"),
+                "The page title does not match the condition");
```
Comment (toleksyn) [inline src/test/java/com/softserveinc/ita/rozetka_test/RozetkaProductNavigationTest.java]: should or shouldn't

File: src/test/java/com/softserveinc/ita/rozetka_test/RozetkaProductNavigationTest.java [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+
+    @BeforeMethod
+    public void openHomepage() {
+        rozetkaHomePage = new HomePage().openHomePage();
+    }
+
+    @Test
+    public void testSelectProductByCatalog() {
+        CategoryPage catalogCategoryItem = rozetkaHomePage
+                .openCategoryPageFromLeftSidebar("Ноутбуки");
```
Comment (toleksyn) [inline src/test/java/com/softserveinc/ita/rozetka_test/RozetkaProductNavigationTest.java]: Ноутбуки -> var

**Review by toleksyn — CHANGES_REQUESTED** [review]

File: src/main/java/com/softserveinc/ita/rozetka/page_objects/FooterPage.java (hunk 2 above)
Comment (toleksyn) [reply src/main/java/com/softserveinc/ita/rozetka/page_objects/FooterPage.java]: // to be extended

File: src/main/java/com/softserveinc/ita/rozetka/page_objects/ProductPage.java [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+
+    public String getCharacteristicDescriptionByIndex(int index) {
+        return $x(String.format("(//ul[@class='characteristics-full__sub-list'])[%d]", index)).text();
+    }
+
+    public String getProductTabsTitle(){
+        return $x("//h2[@class='product-tabs__heading']").text();
+    }
+
+    public List<SelenideElement> getCommentsList(){
```
Comment (toleksyn) [inline src/main/java/com/softserveinc/ita/rozetka/page_objects/ProductPage.java]: List<String> 
$$x().shouldBe(visible).texts();

File: src/test/java/com/softserveinc/ita/rozetka_test/OrderTest.java [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+        int orderQuantity;
+        int orderSum;
+        openOrderPage.editItemInBasket();
+        openBasketPage = new BasketPage();
+        orderSum = openBasketPage.getOrderItemPrice();
+        openBasketPage.pushPlusItem();
+        openBasketPage.openOrderPage();
+        openOrderPage.editItemInBasket();
+        openBasketPage.pushMinusItem();
+        openBasketPage.openOrderPage();
```
Comment (toleksyn) [inline src/test/java/com/softserveinc/ita/rozetka_test/OrderTest.java]: assert

File: src/test/java/com/softserveinc/ita/rozetka_test/RozetkaProductNavigationTest.java [hunk 11]
```diff
@@ -0,0 +1,72 @@
...
+                .openProductByNumber(1)
+                .getProductTitle();
+        Assert.assertTrue(productTittle.toLowerCase().contains("гаманець"), "Search request lead to the wrong product");
+    }
+
+    @Test
+    public void testPossibilityViewingProductDescriptions() {
+        ProductPage productPage = rozetkaHomePage.openCategoryPageFromLeftSidebar("Ноутбуки")
+                .openProductsListPage("Планшети")
+                .openFirstProduct().openProductTabByName("Характеристики");
```
Comment (toleksyn) [inline src/test/java/com/softserveinc/ita/rozetka_test/RozetkaProductNavigationTest.java]: var.method()
     .method() 
     .method();

File: src/main/java/com/softserveinc/ita/rozetka/page_objects/FooterPage.java (hunk 2 above)
Comment (VladimirPetrat) [reply src/main/java/com/softserveinc/ita/rozetka/page_objects/FooterPage.java]: Added comment

File: src/main/java/com/softserveinc/ita/rozetka/page_objects/BasketPage.java [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+    public OrderPage openOrderPage() {
+        $x("//*[@class='button button_size_large button_color_green cart-receipt__submit']").click();
+        return new OrderPage();
+    }
+
+    public void pushPlusItem() {
+        $x("(//*[@class='button button_color_white button_size_medium cart-counter__button'])[2]").click();
+    }
+
+    public void pushMinusItem() {
```
Comment (SerhiiHladkyi) [inline src/main/java/com/softserveinc/ita/rozetka/page_objects/BasketPage.java]: decreaseProductQuantity()

File: src/main/java/com/softserveinc/ita/rozetka/page_objects/CategoryPage.java [hunk 13]
```diff
@@ -0,0 +1,33 @@
+package com.softserveinc.ita.rozetka.page_objects;
+
+import static com.codeborne.selenide.Selenide.$$x;
+import static com.codeborne.selenide.Selenide.$x;
+
+public class CategoryPage extends BasePage {
+
+    public CategoryPage openProductCategoryPage(int index) {
```
Comment (SerhiiHladkyi) [inline src/main/java/com/softserveinc/ita/rozetka/page_objects/CategoryPage.java]: openProductCategoryPage(String categoryName)

File: src/main/java/com/softserveinc/ita/rozetka/page_objects/CategoryPage.java [hunk 14]
```diff
@@ -0,0 +1,33 @@
...
+        $$x("//a[@class='tile-cats__heading tile-cats__heading_type_center']").get(index).click();
+        return new CategoryPage();
+    }
+
+    public ProductPage openProductByNumber(int number) {
+        $$x("//span[@class='goods-tile__title']").get(number).click();
+        return new ProductPage();
+    }
+
+    public ProductsListPage openProductsListPage(int index) {
```
Comment (SerhiiHladkyi) [inline src/main/java/com/softserveinc/ita/rozetka/page_objects/CategoryPage.java:18]: by name?

File: src/main/java/com/softserveinc/ita/rozetka/page_objects/HomePage.java [hunk 15]
```diff
@@ -0,0 +1,31 @@
...
+        open("https://rozetka.com.ua/ua/");
+        return this;
+    }
+
+    public CategoryPage openCategoryPage(int number) {
+        $$x("//a[@class = 'menu-categories__link']").get(number).click();
+        return new CategoryPage();
+    }
+
+    public ProductPage navigateToFirstProduct() {
```
Comment (SerhiiHladkyi) [inline src/main/java/com/softserveinc/ita/rozetka/page_objects/HomePage.java]: open(1)

File: src/main/java/com/softserveinc/ita/rozetka/page_objects/OrderPage.java [hunk 16]
```diff
@@ -0,0 +1,15 @@
+package com.softserveinc.ita.rozetka.page_objects;
+
+import static com.codeborne.selenide.Selenide.$x;
+
+public class OrderPage {
+
+    public BasketPage editItemInBasket() {
```
Comment (SerhiiHladkyi) [inline src/main/java/com/softserveinc/ita/rozetka/page_objects/OrderPage.java]: naming

File: src/main/java/com/softserveinc/ita/rozetka/page_objects/OrderPage.java [hunk 17]
```diff
@@ -0,0 +1,15 @@
...
+        $x("//*[@class='button button_with_icon button_type_link checkout-product__edit-button']").click();
+        return new BasketPage();
+    }
+
+    public boolean isThisIsOrderPage() {
```
Comment (SerhiiHladkyi) [inline src/main/java/com/softserveinc/ita/rozetka/page_objects/OrderPage.java]: boolean method correct in PO? getPageTitle() and then check in test method

File: src/main/java/com/softserveinc/ita/rozetka/page_objects/ProductPage.java [hunk 18]
```diff
@@ -0,0 +1,52 @@
...
+        $x("//button[@class='buy-button button button_with_icon button_color_green button_size_large']").hover().click();
+        return new BasketPage();
+    }
+
+    public ProductPage openProductTabByName(String tabName) {
+        $x(String.format("//a[@class='tabs__link' and contains(text(),'%s')]", tabName)).click();
+        return this;
+    }
+
+    public String getCharacteristicDescriptionByIndex(int index) {
```
Comment (SerhiiHladkyi) [inline src/main/java/com/softserveinc/ita/rozetka/page_objects/ProductPage.java]: by name?

File: src/main/java/com/softserveinc/ita/rozetka/page_objects/BasketPage.java [hunk 19]
```diff
@@ -0,0 +1,39 @@
...
+        $x("(//*[@class='button button_color_white button_size_medium cart-counter__button'])[1]").click();
+        return this;
+    }
+
+    public int getProductCount() {
+        $x("//*[@class='cart-counter__input ng-untouched ng-pristine ng-valid']").click();
+        return Integer.parseInt($x("//*[@class='cart-counter__input ng-untouched ng-pristine ng-valid']").shouldBe(not(Condition.empty)).val());
+    }
+
+    public int getOrderProductPrice() {
```
Comment (toleksyn) [inline src/main/java/com/softserveinc/ita/rozetka/page_objects/BasketPage.java]: what if more than one product?

File: src/main/java/com/softserveinc/ita/rozetka/page_objects/BasketPage.java [hunk 20]
```diff
@@ -0,0 +1,39 @@
...
+    public String getProductTitleByName(String productName) {
+        return $x(String.format("//a[@class='cart-product__title' and contains(text(), '%s')]", productName)).text();
+    }
+
+    public OrderPage openOrderPage() {
+        $x("//*[@class='button button_size_large button_color_green cart-receipt__submit']").click();
+        return new OrderPage();
+    }
+
+    public BasketPage increaseProductCount() {
```
Comment (toleksyn) [inline src/main/java/com/softserveinc/ita/rozetka/page_objects/BasketPage.java]: what if more than one product?

File: src/main/java/com/softserveinc/ita/rozetka/page_objects/HomePage.java [hunk 21]
```diff
@@ -0,0 +1,31 @@
+package com.softserveinc.ita.rozetka.page_objects;
+
+import static com.codeborne.selenide.Selenide.*;
+
+public class HomePage extends BasePage {
+
+    private LeftSidebar leftSidebar;
+
+    public HomePage() {
+        leftSidebar = new LeftSidebar();
```
Comment (toleksyn) [inline src/main/java/com/softserveinc/ita/rozetka/page_objects/HomePage.java]: you can just do that on line 7

File: src/main/java/com/softserveinc/ita/rozetka/page_objects/LeftSidebar.java [hunk 22]
```diff
@@ -0,0 +1,14 @@
+package com.softserveinc.ita.rozetka.page_objects;
+
+import static com.codeborne.selenide.Selenide.$x;
+
+public class LeftSidebar {
+
+    public CategoryPage openCategory(String categoryName){
+        $x(String.format("//a[@class = 'menu-categories__link' and contains(text(), '%s')]", categoryName))
+                .click();
+        $x("//button[@class='search-form__microphone']")
```
Comment (toleksyn) [inline src/main/java/com/softserveinc/ita/rozetka/page_objects/LeftSidebar.java]: is this valid?

File: src/main/java/com/softserveinc/ita/rozetka/page_objects/OrderPage.java [hunk 23]
```diff
@@ -0,0 +1,36 @@
...
+        $x("//div[@class='form__row js-surname']/input").setValue(surname);
+        return this;
+    }
+
+    public OrderPage setName(String name) {
+        $x("//div[@class='form__row js-name']/input").setValue(name);
+        return this;
+    }
+
+    public BasketPage startEditingProductInBasket() {
```
Comment (toleksyn) [inline src/main/java/com/softserveinc/ita/rozetka/page_objects/OrderPage.java]: what if more than one product?

File: src/main/java/com/softserveinc/ita/rozetka/page_objects/ProductPage.java [hunk 24]
```diff
@@ -0,0 +1,50 @@
...
+import static com.codeborne.selenide.Selenide.$x;
+
+public class ProductPage extends BasePage {
+
+    public String getProductTitle() {
+        return $x("//h1[@class='product__title']").text();
+    }
+
+    public ElementsCollection getListOfProductTabs() {
+        return $$x("//a[@class='tabs__link']").shouldBe(CollectionCondition.sizeGreaterThan(4));
```
Comment (toleksyn) [inline src/main/java/com/softserveinc/ita/rozetka/page_objects/ProductPage.java]: shouldHave

File: src/main/java/com/softserveinc/ita/rozetka/page_objects/ProductPage.java [hunk 25]
```diff
@@ -0,0 +1,50 @@
...
+    public String getProductTabsTitle() {
+        return $x("//h2[@class='product-tabs__heading']").text();
+    }
+
+    public ElementsCollection getQuestionList(int amountQuestionsItem) {
+        return $$x("//div[@class='comment']")
+                .shouldBe(CollectionCondition.sizeGreaterThan(amountQuestionsItem));
+    }
+
+    public ElementsCollection getCharacteristicListSize(int amountCharacteristicsItem) {
```
Comment (toleksyn) [inline src/main/java/com/softserveinc/ita/rozetka/page_objects/ProductPage.java]: rename

File: src/main/java/com/softserveinc/ita/rozetka/page_objects/ProductsListPage.java [hunk 26]
```diff
@@ -0,0 +1,55 @@
...
+    public ProductsListPage setSortingType(String sortingTypeKey) {
+        $x("//select").selectOptionContainingText(sortingTypeKey);
+        return this;
+    }
+
+    public String getProductNameByNumber(int number) {
+        return $x(String.format("(//span[@class='goods-tile__title'])[%d]", number)).text();
+    }
+
+    public int getProductPriseByNumber(int number) {
```
Comment (toleksyn) [inline src/main/java/com/softserveinc/ita/rozetka/page_objects/ProductsListPage.java]: rename

File: src/main/java/com/softserveinc/ita/rozetka/page_objects/ProductsListPage.java [hunk 27]
```diff
@@ -0,0 +1,55 @@
...
+    public ProductsListPage openPreviousResultPage() {
+        $x("//a[@class='button button_color_gray button_size_medium pagination__direction']").click();
+        return this;
+    }
+
+    public int getCurrentPageNumber() {
+        return Integer.parseInt($x("//a[@class='pagination__link pagination__link_state_active']").text());
+    }
+
+    public ProductsListPage useFilter(String filterName) {
```
Comment (toleksyn) [inline src/main/java/com/softserveinc/ita/rozetka/page_objects/ProductsListPage.java]: setFilter().filter()

File: src/test/java/com/softserveinc/ita/rozetka_test/OrderTest.java [hunk 28]
```diff
@@ -0,0 +1,23 @@
...
+
+    @BeforeMethod
+    public void addProductToBasketOpenOrder() {
+        openOrderPage = new HomePage()
+                .openHomePage()
+                .openProductByNumber(1)
+                .addProductToBasket()
+                .openOrderPage();
+    }
+}
```
Comment (toleksyn) [inline src/test/java/com/softserveinc/ita/rozetka_test/OrderTest.java:24]: where's the test? add comment or todo

File: src/test/java/com/softserveinc/ita/rozetka_test/RozetkaProductNavigationTest.java [hunk 29]
```diff
@@ -0,0 +1,16 @@
...
+public class RozetkaProductNavigationTest extends TestRunner {
+
+    private HomePage rozetkaHomePage;
+
+    @BeforeMethod
+    public void openHomepage() {
+        rozetkaHomePage = new HomePage().openHomePage();
+    }
+}
+
```
Comment (toleksyn) [inline src/test/java/com/softserveinc/ita/rozetka_test/RozetkaProductNavigationTest.java:20]: same

**Review by toleksyn — CHANGES_REQUESTED** [review]

File: src/main/java/com/softserveinc/ita/rozetka/page_objects/ProductPage.java [hunk 30]
```diff
@@ -0,0 +1,50 @@
...
+    public ProductPage openProductTabByName(String tabName) {
+        $x(String.format("//a[@class='tabs__link' and contains(text(),'%s')]", tabName)).click();
+        return this;
+    }
+
+    public String getProductTabsTitle() {
+        return $x("//h2[@class='product-tabs__heading']").text();
+    }
+
+    public ElementsCollection getQuestionList(int amountQuestionsItem) {
```
Comment (toleksyn) [inline src/main/java/com/softserveinc/ita/rozetka/page_objects/ProductPage.java]: re-do this into returning List<String

File: src/main/java/com/softserveinc/ita/rozetka/page_objects/ProductPage.java [hunk 31]
```diff
@@ -0,0 +1,50 @@
...
+
+    public ElementsCollection getListOfProductTabs() {
+        return $$x("//a[@class='tabs__link']").shouldBe(CollectionCondition.sizeGreaterThan(4));
+    }
+
+    public SelenideElement getViewedProduct(int number) {
+        return $x(String.format("(//section[@class='recently-viewed']//a[@class='lite-tile__title'])[%d]", number));
+    }
+
+    public CategoryPage returnToCategoryPageByName(String categoryName) {
```
Comment (toleksyn) [inline src/main/java/com/softserveinc/ita/rozetka/page_objects/ProductPage.java]: open*()

**Review by toleksyn — CHANGES_REQUESTED** [review]

File: src/test/java/com/softserveinc/ita/rozetka_test/RozetkaProductNavigationTest.java (hunk 29 above)
Comment (VladimirPetrat) [reply src/test/java/com/softserveinc/ita/rozetka_test/RozetkaProductNavigationTest.java:20]: Added comment

File: src/test/java/com/softserveinc/ita/rozetka_test/OrderTest.java (hunk 28 above)
Comment (VladimirPetrat) [reply src/test/java/com/softserveinc/ita/rozetka_test/OrderTest.java:24]: Added comment

File: src/main/java/com/softserveinc/ita/rozetka/page_objects/HomePage.java (hunk 3 above)
Comment (VladimirPetrat) [reply src/main/java/com/softserveinc/ita/rozetka/page_objects/HomePage.java:16]: Outdated

File: src/test/java/com/softserveinc/ita/rozetka_test/OrderTest.java [hunk 32]
```diff
@@ -0,0 +1,29 @@
...
+import org.testng.annotations.Test;
+
+public class OrderTest extends TestRunner {
+
+    private OrderPage openOrderPage;
+    private BasketPage openBasketPage;
+    private String searchString;
+
+    public OrderTest() {
+    }
```
Comment (toleksyn) [inline src/test/java/com/softserveinc/ita/rozetka_test/OrderTest.java]: why is this needed?

File: src/main/java/com/softserveinc/ita/rozetka/page_objects/BasketPage.java [hunk 33]
```diff
@@ -0,0 +1,43 @@
...
+    public String getProductTitleByName(String productName) {
+        return $x(String.format("//a[@class='cart-product__title' and contains(text(), '%s')]", productName)).text();
+    }
+
+    public OrderPage openOrderPage() {
+        $x("//*[@class='button button_size_large button_color_green cart-receipt__submit']").click();
+        return new OrderPage();
+    }
+
+    public BasketPage increaseProductCount(int number) {
```
Comment (toleksyn) [inline src/main/java/com/softserveinc/ita/rozetka/page_objects/BasketPage.java]: number - number to increase or number or product?

File: src/main/java/com/softserveinc/ita/rozetka/page_objects/OrderPage.java [hunk 34]
```diff
@@ -0,0 +1,36 @@
...
+        $x("//div[@class='form__row js-surname']/input").setValue(surname);
+        return this;
+    }
+
+    public OrderPage setName(String name) {
+        $x("//div[@class='form__row js-name']/input").setValue(name);
+        return this;
+    }
+
+    public BasketPage startEditingProductsInBasket() {
```
Comment (toleksyn) [inline src/main/java/com/softserveinc/ita/rozetka/page_objects/OrderPage.java:17]: still invalid, what product will be edited if multiples products there?

File: src/main/java/com/softserveinc/ita/rozetka/page_objects/ProductPage.java [hunk 35]
```diff
@@ -0,0 +1,53 @@
...
+        return $x("//h2[@class='product-tabs__heading']").text();
+    }
+
+    public int getQuestionListSize(int amountQuestionsItem) {
+        return $$x("//div[@class='comment']")
+                .shouldHave(CollectionCondition.sizeGreaterThan(amountQuestionsItem))
+                .size();
+    }
+
+    public List<String> getCharacteristicsTexts(int amountCharacteristicsItem) {
```
Comment (toleksyn) [inline src/main/java/com/softserveinc/ita/rozetka/page_objects/ProductPage.java]: no need to pass amountCharacteristicsItem as parameter
just use in method

File: src/main/java/com/softserveinc/ita/rozetka/page_objects/ProductsListPage.java [hunk 36]
```diff
@@ -0,0 +1,59 @@
...
+
+import com.codeborne.selenide.ElementsCollection;
+import com.codeborne.selenide.Selenide;
+
+import static com.codeborne.selenide.Selenide.$x;
+
+public class ProductsListPage extends BasePage {
+
+    public ElementsCollection getProductList() {
+        return Selenide.$$x("//span[@class='goods-tile__title']");
```
Comment (toleksyn) [inline src/main/java/com/softserveinc/ita/rozetka/page_objects/ProductsListPage.java]: no waits here

File: src/main/java/com/softserveinc/ita/rozetka/page_objects/ProductsListPage.java [hunk 37]
```diff
@@ -0,0 +1,59 @@
...
+        $x(String.format("(//a[@class='goods-tile__picture'])[%d]", number)).click();
+        return new ProductPage();
+    }
+
+    public ProductsListPage setSortingType(String sortingTypeKey) {
+        $x("//select").selectOptionContainingText(sortingTypeKey);
+        return this;
+    }
+
+    public String getProductNameByNumber(int number) {
```
Comment (toleksyn) [inline src/main/java/com/softserveinc/ita/rozetka/page_objects/ProductsListPage.java]: getProductNameForProduct(1)

File: src/main/java/com/softserveinc/ita/rozetka/page_objects/ProductsListPage.java [hunk 38]
```diff
@@ -0,0 +1,59 @@
...
+        return $x(String.format("(//span[@class='goods-tile__title'])[%d]", number)).text();
+    }
+
+    public int getProductPriceByNumber(int number) {
```
Comment (toleksyn) [inline src/main/java/com/softserveinc/ita/rozetka/page_objects/ProductsListPage.java]: getProductPriceForProduct(1)

File: src/main/java/com/softserveinc/ita/rozetka/page_objects/ProductsListPage.java [hunk 39]
```diff
@@ -0,0 +1,59 @@
...
+        $x("//a[@class='button button_color_gray button_size_medium" +
+                " pagination__direction']").click();
+        return this;
+    }
+
+    public int getCurrentPageNumber() {
+        return Integer.parseInt($x("//a[@class='pagination__link pagination__link_state_active']").text());
+    }
+
+    public ProductsListPage filter(String filterName) {
```
Comment (toleksyn) [inline src/main/java/com/softserveinc/ita/rozetka/page_objects/ProductsListPage.java]: does this filter or just selects filter option?

**Review by toleksyn — CHANGES_REQUESTED** [review]

File: src/main/java/com/softserveinc/ita/rozetka/page_objects/BasketPage.java (hunk 33 above)
Comment (kuguk) [reply src/main/java/com/softserveinc/ita/rozetka/page_objects/BasketPage.java]: method increases the count of specific product selected by its number in the basket
variable (int number) changed to (int numberOfProduct)

File: src/main/java/com/softserveinc/ita/rozetka/page_objects/OrderPage.java (hunk 34 above)
Comment (kuguk) [reply src/main/java/com/softserveinc/ita/rozetka/page_objects/OrderPage.java:17]: the task of this method is to start editing
It opens Basket page for editing.  Item(s) of product(s) is/are chosen by user in another method(s)

File: src/test/java/com/softserveinc/ita/rozetka_test/OrderTest.java (hunk 32 above)
Comment (kuguk) [reply src/test/java/com/softserveinc/ita/rozetka_test/OrderTest.java]: fixed

File: src/main/java/com/softserveinc/ita/rozetka/page_objects/OrderPage.java (hunk 34 above)
Comment (toleksyn) [reply src/main/java/com/softserveinc/ita/rozetka/page_objects/OrderPage.java:17]: use singular then -> startEditingProductInBasket()

File: src/main/java/com/softserveinc/ita/rozetka/page_objects/ProductsListPage.java [hunk 40]
```diff
@@ -0,0 +1,61 @@
...
+        return $x(String.format("(//span[@class='goods-tile__title'])[%d]", number)).text();
+    }
+
+    public int getProductPriceForProduct(int number) {
+        return Integer.parseInt($x(String.format("(//span[@class='goods-tile__price-value'])[%d]", number))
+                .text()
+                .replaceAll(" ", ""));
+    }
+
+    public ProductsListPage openNextResultPage() {
```
Comment (toleksyn) [inline src/main/java/com/softserveinc/ita/rozetka/page_objects/ProductsListPage.java]: openNextPage()

File: src/main/java/com/softserveinc/ita/rozetka/page_objects/ProductsListPage.java [hunk 41]
```diff
@@ -0,0 +1,61 @@
...
+        $x("//a[@class='button button_color_gray button_size_medium" +
+                " pagination__direction pagination__direction_type_forward']").click();
+        return this;
+    }
+
+    public ProductsListPage openPreviousResultPage() {
```
Comment (toleksyn) [inline src/main/java/com/softserveinc/ita/rozetka/page_objects/ProductsListPage.java]: openPreviousPage()

**Review by toleksyn — CHANGES_REQUESTED** [review]
